### PR TITLE
Adding quality metrics and opening discussion

### PR DESCRIPTION
**Pull request by alexdesiqueira**
## Description

This brings us a basic implementation for several metrics: precision, recall, specificity, accuracy, and the coefficients of Matthews, Dice (F1 score), and informedness. Open to discussion (see first comment).

## Checklist

- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/) (I think so)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by alexandrejaguar** [comment]
Hi @scikit-image/core,
I implemented these functions a while ago, and since we don't have them, I'd like to discuss their addition. For that, some questions:
1. Do we need these metrics? I think scikit-learn has some of them. Please feel free to close the PR if they are not worth the effort.
Is the PR still alive? OK, the other issues:
2. They are scattered around one file. I think they maybe would be better implemented as one class instead, that could also return a confusion matrix. Thoughts?
3. This implementation only works on binary images right now. I am not working on cases with labels, or fancier images; that would need more effort. sklearn has their confusion matrix as something like that, however (it works on several classes). Maybe it could be a start.
After discussing this I'd work on documentation and all.
Thank you for your time.

**Comment by alexandrejaguar** [comment]
> 3. This implementation only works on binary images right now. I am not working on cases with labels, or fancier images; that would need more effort. sklearn has their confusion matrix as something like that, however (it works on several classes). Maybe it could be a start.

On the other hand, some of these measures only work on binary images... silly me :relieved:

**Comment by stefanv** [comment]
There seems to be a few segmentation/clustering comparison algorithms out there, including Normalized Probabilistic Rand: https://ieeexplore.ieee.org/abstract/document/1565332.  I guess there are others too; we should try to find a review paper.

**Comment by alexandrejaguar** [comment]
@stefanv I have one or two reviews I used during previous research; I'll paste their names and possible links soon.

**Comment by stefanv** [comment]
Quick question: some of these features seem to intersect with
scikit-learn.  Are the functions in
https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics
not usable for our images?

**Comment by alexandrejaguar** [comment]
I missed it :man_facepalming: I saw only `confusion_matrix` before. I believe they all are usable for our images, and I think they're prepared to deal with more than two classes.

**Comment by stefanv** [comment]
Are there any useful ones we can add?

**Comment by alexandrejaguar** [comment]
Closed due to overlapping with `scikit-learn` functions. Please reopen if necessary.

**Comment by jni** [comment]
I think I'd like this reopened. I'd also appreciate reviews on #3354!

**Comment by alexandrejaguar** [comment]
@jni any opinions on this one?

**Comment by alexdesiqueira** [comment]
Reopening this; @jni, would you like to discuss that?

**Comment by matthew-brett** [comment]
We have changed the base branch for this PR from the `main` branch to the `pre-megamove-integration` branch.

## Summary - what this means for your PR

You can continue to work on your PR, now based on `pre-megamove-integration`.  You should update your branch to the current state of `pre-megamove-integration` (this is what the Github "Update branch" button does).   Please **do not merge the scikit-image `main` branch into your PR branch**.

Once we've merged your PR into the `pre-megamove-integration` branch, we'll update your changes for Megamove, and merge into the `main` branch.

In addition - our automated script detected that your branch had merge conflicts with the pre-Megamove branch. Please make sure to merge in the `pre-megamove-integration` branch, and resolve the merge conflicts in the way that you want.

Read on for the Megamove rationale, and more detailed instructions.

<details>

## Rationale

We've changed the base branch to make it easier and smoother to continue to work on your PR, while we work through a rather large change in the directory structure of the repository - a change that we have called *Megamove*.

See [the Megamove explanation issue](https://github.com/scikit-image/scikit-image/issues/8232) for more detail.

Changing the base branch means you do not see merge conflicts related to the *Megamove* change, and can continue to work with the repository in its original state, without worrying about porting those changes across to the new post-Megamove structure.

As you'll see in the linked issue, you can continue to work on your PR as before, and you can continue to use the Github GUI to "Update branch" - this updates from the new `pre-megamove-integration` branch.   However, there is one caveat: please, do not try and merge the `scikit-image` `main` branch into this one, but instead, keep track of the `pre-megamove-integration` branch.  The issue explains more.

To emphasize again, **do not try and merge the scikit-image `main` branch**.  You will get Megamove related merge conflicts that are rather tricky to resolve correctly.

When we've approved your PR, we'll merge into the `pre-megamove-integration` branch, and then process it so we can merge it into the `main` branch.   We'll let you know when we've merged your PR changes into `main`.

</details>

Any questions, please reply here.
